src/DotNetWorker.Grpc/GrpcWorker.cs

```diff
@@ -279,10 +279,44 @@ private async Task<FunctionMetadataResponse> GetFunctionMetadataAsync(string fun
             {
                 var functionMetadataList = await _functionMetadataProvider.GetFunctionMetadataAsync(functionAppDirectory);
 
-                foreach (var func in functionMetadataList)
+                // depending on the FunctionMetadataProvider used, we may get RpcFunctionMetadata instead of a generated IFunctionMetadata object
+                if (functionMetadataList.First().GetType() == typeof(RpcFunctionMetadata))
```

Comment:
Would `functionMetadataList.First() is RpcFunctionMetadata funcAsRpcMetadata` produce same output?